SWLOR.Game.Server/Perk/DarkSide/ForceLeech.cs

```diff
@@ -0,0 +1,175 @@
+        }
+
+        public float CooldownTime(NWPlayer oPC, float baseCooldownTime)
+        {
+            return baseCooldownTime;
+        }
+
+        public void OnImpact(NWPlayer player, NWObject target, int level)
+        {
+            float recoveryPercent;
+            var effectiveStats = _stat.GetPlayerItemEffectiveStats(player);
+            int darkBonus = effectiveStats.DarkAbility;
+            int length;
+            int dotAmount;
+            int wisdom = player.WisdomModifier;
+            int intelligence = player.IntelligenceModifier;
+
+
+            switch (level)
+            {
+                case 1:
+                    length = 6;
+                    dotAmount = 2;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 2:
+                    length = 9;
+                    dotAmount = 2;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 3:
+                    length = 6;
+                    dotAmount = 4;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 4:
+                    length = 6;
+                    dotAmount = 4;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 5:
+                    length = 9;
+                    dotAmount = 4;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 6:
+                    length = 9;
+                    dotAmount = 6;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 7:
+                    length = 12;
+                    dotAmount = 8;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 8:
+                    length = 12;
+                    dotAmount = 8;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 9:
+                    length = 15;
+                    dotAmount = 12;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 10:
+                    length = 15;
+                    dotAmount = 12;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 11: // Only available with background bonus
+                    length = 18;
+                    dotAmount = 12;
+                    recoveryPercent = 1.0f;
+                    break;
+                default: return;
+            }
+
+            _skill.RegisterPCToAllCombatTargetsForSkill(player, SkillType.DarkSideAbilities);
+
+            if (length > 0.0f && dotAmount > 0)
+            {
+                _customEffect.ApplyCustomEffect(player, target.Object, CustomEffectType.ForceLeech, length, level, dotAmount + "," + recoveryPercent);
+
+            }
+
+            _skill.RegisterPCToAllCombatTargetsForSkill(player, SkillType.DarkSideAbilities);
```

Comment:
Again, I think your code is out of sync but I added a target argument to the end of this method. It's to fix an issue where registrations wouldn't work if you used a force ability on a target at the very beginning of battle.

We'll fix this one tomorrow as well.